### ISSUE TITLE
Plugin cause Eclipse freeze of 5s on startup

### ISSUE THREAD

**Issue by bubbleguuum** [de-jcup/egradle#354]
Observed on Eclipse Photon, a UI freeze of 5sec on startup mentionned in log file:

!ENTRY org.eclipse.osgi 2 0 2018-07-22 14:26:37.809
!MESSAGE While loading class "de.jcup.egradle.eclipse.ide.console.EGradleConsolePageParticipant", thread "Thread[main,6,main]" timed out waiting (5009ms) for thread "Thread[Worker-6: Decoration Calculation,5,main]" to finish starting bundle "de.jcup.egradle.eclipse.plugin.ide_2.5.0 [582]". To avoid deadlock, thread "Thread[main,6,main]" is proceeding but "de.jcup.egradle.eclipse.ide.console.EGradleConsolePageParticipant" may not be fully initialized.
!STACK 0
org.osgi.framework.BundleException: Unable to acquire the state change lock for the module: osgi.identity; type="osgi.bundle"; version:Version="2.5.0"; osgi.identity="de.jcup.egradle.eclipse.plugin.ide"; singleton:="true" [id=582] STARTED [STARTED]
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:337)
	at org.eclipse.osgi.container.Module.start(Module.java:401)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:468)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:114)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:505)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:328)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:392)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:470)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:419)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:411)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.loadClass(EquinoxBundle.java:609)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:177)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:931)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:243)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:60)
	at org.eclipse.ui.internal.console.ConsolePageParticipantExtension.createDelegate(ConsolePageParticipantExtension.java:72)
	at org.eclipse.ui.internal.console.ConsoleManager.getPageParticipants(ConsoleManager.java:446)
	at org.eclipse.ui.internal.console.ConsoleView.doCreatePage(ConsoleView.java:327)
	at org.eclipse.ui.part.PageBookView.createPage(PageBookView.java:375)
	at org.eclipse.ui.part.PageBookView.partActivated(PageBookView.java:715)
	at org.eclipse.ui.internal.console.ConsoleView$4.run(ConsoleView.java:411)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:3906)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3537)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
Caused by: java.util.concurrent.TimeoutException: Timeout after waiting 5 seconds to acquire the lock.
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:334)
	... 47 more
Root exception:
java.util.concurrent.TimeoutException: Timeout after waiting 5 seconds to acquire the lock.
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:334)
	at org.eclipse.osgi.container.Module.start(Module.java:401)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:468)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:114)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:505)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:328)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:392)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:470)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:419)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:411)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.loadClass(EquinoxBundle.java:609)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:177)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:931)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:243)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:60)
	at org.eclipse.ui.internal.console.ConsolePageParticipantExtension.createDelegate(ConsolePageParticipantExtension.java:72)
	at org.eclipse.ui.internal.console.ConsoleManager.getPageParticipants(ConsoleManager.java:446)
	at org.eclipse.ui.internal.console.ConsoleView.doCreatePage(ConsoleView.java:327)
	at org.eclipse.ui.part.PageBookView.createPage(PageBookView.java:375)
	at org.eclipse.ui.part.PageBookView.partActivated(PageBookView.java:715)
	at org.eclipse.ui.internal.console.ConsoleView$4.run(ConsoleView.java:411)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:3906)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3537)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1498)

**Comment by bubbleguuum** [comment]
This error (and freeze) went away after I disabled Preferences > General > Appearance > Label decoration > EGradle project decorator.

**Comment by de-jcup** [comment]
Hello @bubbleguuum .
I also use EGradle with photon at my work without any problems.

Could you please tell my which OS your are exactly using and which java version are you running your eclipse?

**Comment by bubbleguuum** [comment]
Using latest Windows 10, Eclipse running on Java 1.8.0_181.
I've re-enabled "EGradle project decorator" an I do not get this error anymore in startup log.
But I removed some plugins since my initial post, so it could be that...
I'd say that if you cannot think of an obvious cause for the stack trace I posted, don't spend too much time on it.

**Comment by de-jcup** [comment]
@bubbleguuum 
The `EGradleConsolePageParticipant` is the special console handler which enables console outputs to have coloured key words on gradle builds. Has nothing to do with the project decorators.

Maybe one of the uninstalled plugins was the reason for the behaviour.

At https://github.com/de-jcup/egradle/blob/9e977db83de78ff5c31d01ba1222778fadcbbcf5/egradle-plugin-ide/src/main/java-eclipse/de/jcup/egradle/eclipse/ide/console/EGradleConsolePageParticipant.java#L58-L64
EGradle console adoption does only happen when inside process console with containing "EGradle" in title output or be dedicated EGradle console itself. So normally this should not influence any other plugins or system start of eclipse.

Please try out if the problem does not appear any longer and if no longer problems please close this issue.

**Comment by bubbleguuum** [comment]
I could reproduce it and narrow when it happens: it only happen if when you exit eclipse the Console tab is active (eg its content visible) AND "EGradle project decorator" enabled in Preferences. Should be easy to reproduce it for you know.

**Comment by de-jcup** [comment]
Hmm.. had something similar (2019-03 RCP at development start, on EGradle 2.6.0):

```
org.osgi.framework.BundleException: Unable to acquire the state change lock for the module: osgi.identity; type="osgi.bundle"; version:Version="2.6.0"; osgi.identity="de.jcup.egradle.eclipse.plugin.ide"; singleton:="true" [id=674] STARTED [STARTED]
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:344)
	at org.eclipse.osgi.container.Module.start(Module.java:408)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:471)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:117)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:557)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:331)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:395)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:473)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:153)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at de.jcup.egradle.eclipse.ide.console.EGradleConsolePageParticipant.init(EGradleConsolePageParticipant.java:59)
	at org.eclipse.ui.internal.console.ConsoleView$3.run(ConsoleView.java:341)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.ui.internal.console.ConsoleView.doCreatePage(ConsoleView.java:338)
	at org.eclipse.ui.part.PageBookView.createPage(PageBookView.java:378)
	at org.eclipse.ui.part.PageBookView.partActivated(PageBookView.java:718)
	at org.eclipse.ui.internal.console.ConsoleView.lambda$0(ConsoleView.java:412)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4906)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4475)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)
Caused by: java.util.concurrent.TimeoutException: Timeout after waiting 5 seconds to acquire the lock.
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:341)
	... 44 more
Root exception:
java.util.concurrent.TimeoutException: Timeout after waiting 5 seconds to acquire the lock.
	at org.eclipse.osgi.container.Module.lockStateChange(Module.java:341)
	at org.eclipse.osgi.container.Module.start(Module.java:408)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:471)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:117)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:557)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:331)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:395)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:473)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:153)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at de.jcup.egradle.eclipse.ide.console.EGradleConsolePageParticipant.init(EGradleConsolePageParticipant.java:59)
	at org.eclipse.ui.internal.console.ConsoleView$3.run(ConsoleView.java:341)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.ui.internal.console.ConsoleView.doCreatePage(ConsoleView.java:338)
	at org.eclipse.ui.part.PageBookView.createPage(PageBookView.java:378)
	at org.eclipse.ui.part.PageBookView.partActivated(PageBookView.java:718)
	at org.eclipse.ui.internal.console.ConsoleView.lambda$0(ConsoleView.java:412)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4906)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4475)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)

```

**Comment by de-jcup** [comment]
Maybe interesting: https://www.eclipse.org/forums/index.php/t/823236/

**Comment by de-jcup** [comment]
When I prepared version to 2.6.1 for development the problem vanished

**Comment by de-jcup** [comment]
Former comment is not true - still same problems at development

**Comment by de-jcup** [comment]
Tried to change EGradleConsolePageParticipant to do nothing, but its still the same 5s freeze.
Tried to stop loading XML did also not work.

**Comment by de-jcup** [comment]
Another issue with this - was slowing down 2019-03 while testing out ASP server approach in asciidoctor-editor - it was slowingdown 30 seconds!!!!! MUST be fixed!

`While loading class "de.jcup.egradle.eclipse.ide.console.EGradleSystemConsole", thread "Thread[main,6,main]" timed out waiting (30211ms) for thread "Thread[Worker-0: Decoration Calculation,5,main]" to finish starting bundle "de.jcup.egradle.eclipse.plugin.ide_2.6.0 [1069]". To avoid deadlock, thread "Thread[main,6,main]" is proceeding but "de.jcup.egradle.eclipse.ide.console.EGradleSystemConsole" may not be fully initialized.`

**Comment by de-jcup** [comment]
2019-06 is freezing also 30 seconds!!!!
EGradleConsolePageParticipant init seems to be the problem:

```
ThreadId: 1 ThreadName: main ThreadState: RUNNABLE
  Blocked On: org.eclipse.osgi.internal.container.EquinoxReentrantLock@35c646b5[Locked by thread Worker-4: Decoration Calculation]
  Synchronizers Locked: none
  Monitors Locked: 
    org.eclipse.swt.widgets.RunnableLock@720cbb7d
  Stack Trace: 
    sun.management.ThreadImpl.dumpThreads0(Native Method)
    ...
    org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:155)
    java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    de.jcup.egradle.eclipse.ide.console.EGradleConsolePageParticipant.init(EGradleConsolePageParticipant.java:59)
    org.eclipse.ui.internal.console.ConsoleView$3.run(ConsoleView.java:341)

```

funny: line 59 is only
`needsEGradleStyling = needsEGradleStyling || console instanceof EGradleSystemConsole;`
so having here a relation/reference to EGradleSystemConsole which coulde enforce class loading of 
EGradleSystemConsole at this time. Maybe this is the point.

Because having  another  check at next line:
```
needsEGradleStyling = needsEGradleStyling
					|| ("org.eclipse.debug.ui.ProcessConsoleType".equals(type) && name.indexOf("EGradle") != -1);
```
the removement of the line 59 shall have no logical effect, but could solve the problem.
